knowledge-base/set-report-parameter-date-range.md

```diff
@@ -0,0 +1,148 @@
+---
+title: Restrict report date parameters to a specific range
+description: How to add a max range to a DateTime Report Parameter
+type: how-to
+page_title: Limit the date selection range of a DateTime Report Parameter 
+slug: set-report-parameter-date-range
+position: 
+tags: 
+ticketid: 1590408
+res_type: kb
+---
+
+## Environment
+<table>
+	<tbody>
+		<tr>
+			<td>Product</td>
+			<td>Progress Telerik Reporting</td>
+		</tr>
+		<tr>
+			<td>Report Viewer</td>
+			<td>HTML5-based Report Viewers</td>
+		</tr>
+	</tbody>
+</table>
+
+
+## Description
+
+In some scenarios, one might want to add set the [Max](https://docs.telerik.com/kendo-ui/api/javascript/ui/datepicker/configuration/max) date of a `DateTime` report parameter based on the selected date of another `DateTime` report parameter, essentially creating a range of dates that an end-user can pick from.
+
+Let's imagine a report that has two `DateTime` Report Parameters with the following settings:
+
+- The first report parameter will have its [Name](/reporting/api/Telerik.Reporting.ReportParameter#Telerik_Reporting_ReportParameter_Name) set to the value `DateRangeParam1` while its [Text](/reporting/api/Telerik.Reporting.ReportParameter#Telerik_Reporting_ReportParameter_Text) property can be equal to `First date` 
+
+- The second report parameter will have its [Name](/reporting/api/Telerik.Reporting.ReportParameter#Telerik_Reporting_ReportParameter_Name) set to the value `DateRangeParam2` while its [Text](/reporting/api/Telerik.Reporting.ReportParameter#Telerik_Reporting_ReportParameter_Text) property can be equal to `Second date`
+
+>note In order to be able to use the code below without making changes to it, your report parameter should have their __Names__ and __Texts__ as clarified above
+
+In order to add a range between those two DateTime report parameters, it is necessary to have access to the current value of the `Start date` report parameter. That is because the `max date` of the second report parameter will depend on the value of the first. 
+
+The only to have access to the value of the first parameter every time that it changes is to create our own parameter editors as shown in the [How to Create a Custom Parameter Editor]({%slug telerikreporting/using-reports-in-applications/display-reports-in-applications/web-application/html5-report-viewer/customizing/how-to-create-a-custom-parameter-editor%}) article. We can re-use the kendo [DatePIcker](https://demos.telerik.com/kendo-ui/datepicker/index) since it comes with the ability to specify a [max](https://docs.telerik.com/kendo-ui/api/javascript/ui/datepicker/methods/max) date out of the box.
+
+## Solution
+
+In the end, the code for the custom `match` function and the custom `createEditor` functions may look as follows:
```

Comment:
Maybe, emphasize that the reader should take a closer look at the **How to Create a Custom Parameter Editor** KB linked above **first** if they are not sure what the `match` and `createEditor` functions stand for.